packages/app/updater.ts

```diff
@@ -12,11 +13,29 @@ class AppUpdater {
 		log.transports.file.level = is.dev ? "info" : "error";
 		autoUpdater.logger = log;
 
-		autoUpdater.checkForUpdatesAndNotify();
+		// Check if notifications should be shown and set up event handlers
+		if (config.get("updates.showNotifications")) {
```

Comment:
remove the event handlers. simply call autoUpdater.checkForUpdates() or autoUpdater.checkForUpdatesAndNotify() depending on the config value.